base/applications/cmdutils/tasklist/tasklist.c

```diff
@@ -0,0 +1,338 @@
+ * COPYRIGHT:   Copyright 2020 He Yang (1160386205@qq.com)
+ */
+
+#include "tasklist.h"
+
+// the strings in opList are the command-line options.
+// should always correspond with the defines below, in sequence (except OP_PARAM_INVALID)
+static PCWSTR opList[] = { L"?", L"nh" };
+
+#define OP_PARAM_INVALID    -1
+#define OP_PARAM_HELP       0
+#define OP_PARAM_NOHEADER    1
+
+// the max string length PrintResString can handle
+#define RES_STR_MAXLEN 64
+
+// Print split line
+VOID PrintSplitLine(UINT Length)
+{
+    for (; Length; Length--)
+    {
+        ConPuts(StdOut, L"=");
+    }
+}
+
+// Print spaces
+VOID PrintSpace(UINT Length)
+{
+    ConPrintf(StdOut, L"%*ls", (INT)Length, L"");
+}
+
+// Print a string.
+// if bAlignLeft == TRUE then aligned to left, otherwise aligned to right
+// MaxWidth is the width for printing.
+INT PrintString(LPCWSTR String, INT MaxWidth, BOOL bAlignLeft)
+{
+    return ConPrintf(StdOut, bAlignLeft ? L"%-*.*ls" : L"%*.*ls", MaxWidth, MaxWidth, String);
+}
+
+// Print a string from resource
+// if bAlignLeft == TRUE then aligned to left, otherwise aligned to right
+// MaxWidth is the width for printing.
+// The string WILL be truncated if it's longer than RES_STR_MAXLEN
+INT PrintResString(HINSTANCE hInstance, UINT uid, INT MaxWidth, BOOL bAlignLeft)
+{
+    if (!hInstance)
+        return 0;
+    
+    WCHAR StringBuffer[RES_STR_MAXLEN];
+    LoadStringW(hInstance, uid, StringBuffer, _countof(StringBuffer));
+    return PrintString(StringBuffer, MaxWidth, bAlignLeft);
+}
+
+// Print a number, aligned to right.
+// MaxWidth is the width for printing.
+// the number WILL NOT be truncated if it's longer than MaxWidth
+VOID PrintNum(LONGLONG Number, INT MaxWidth)
+{
+    ConPrintf(StdOut, L"%*lld", MaxWidth, Number);
+}
+
+// Print memory size using KB as unit, with comma-separated number, aligned to right.
+// MaxWidth is the width for printing.
+// the number WILL be truncated if it's longer than MaxWidth
+BOOL PrintMemory(SIZE_T MemorySizeByte, INT MaxWidth, HINSTANCE hInstance)
+{
+    if (MaxWidth <= 0 || !hInstance)
+        return FALSE;
+
+    SIZE_T MemorySize = MemorySizeByte >> 10;
+
+    WCHAR NumberString[27] = { 0 }; // length 26 is enough to display ULLONG_MAX in decimal with comma, one more for zero-terminated.
+    C_ASSERT(sizeof(SIZE_T) <= 8);
+
+    PWCHAR pNumberStr = NumberString;
+
+    // calculate the length
+    INT PrintLength = 0;
+    SIZE_T Tmp = MemorySize;
+    do
+    {
+        Tmp /= 10;
+        PrintLength += 1;
+    } while (Tmp);
+
+    INT Mod = 1;
+    for (INT i = 0; i < PrintLength - 1; i++)
+    {
+        Mod *= 10;
+    }
+
+    for (INT i = PrintLength - 1; i >= 0; i--)
+    {
+        *pNumberStr = L'0' + (MemorySize / Mod);
+        MemorySize %= Mod;
+        pNumberStr++;
+        
+        if (i && i % 3 == 0)
+        {
+            *pNumberStr = L',';
+            pNumberStr++;
+        }
+        Mod /= 10;
+    }
```

Comment:
By the way I want to mention something: This formatting is also locale-dependent. For example when running in French locale:
![image](https://user-images.githubusercontent.com/1969829/108563790-59e15300-7302-11eb-8843-0852e9db6147.png)
Note: No comma for thousand delimiters, and the unit is indeed localized (re. our past discussion, so I think that if in tasklist resources they just have "%s K" it may be a fallback)
See:
https://docs.microsoft.com/en-us/globalization/locale/number-formatting-in-win32
https://docs.microsoft.com/en-us/windows/win32/api/winnls/nf-winnls-getnumberformatw
and
https://git.reactos.org/?p=reactos.git;a=blob;f=base/applications/mscutils/eventvwr/eventvwr.c;hb=ea26bef01de20d5c140a426397627089892356a0#l1113

Reply:
If this looks too complicated this could wait for a separate PR.

Reply:
I'm also planning to add other format support later (`/fo` option), possibly I can fix this in that PR together🤣

Reply:
At that stage yes a separate PR for the other new features would be better.